src/main/java/io/confluent/kafkarest/MetadataObserver.java

```diff
@@ -180,20 +185,21 @@ public int getLeaderId(final String topicName, final int partitionId) {
 
       Partition p = new Partition();
       p.setPartition(partId);
-      LeaderAndIsr leaderAndIsr =
-          zkUtils.getLeaderAndIsrForPartition(topic, partId).get();
-      p.setLeader(leaderAndIsr.leader());
-      scala.collection.immutable.Set<Integer> isr = leaderAndIsr.isr().toSet();
-      List<PartitionReplica> partReplicas = new Vector<PartitionReplica>();
-      for (Object brokerObj : JavaConversions.asJavaCollection(part.getValue())) {
-        int broker = (Integer) brokerObj;
-        PartitionReplica
-            r =
-            new PartitionReplica(broker, (leaderAndIsr.leader() == broker), isr.contains(broker));
-        partReplicas.add(r);
+      if(!zkUtils.getLeaderAndIsrForPartition(topic, partId).isEmpty()) {
```

Comment:
In the other cases, you looked up the items twice. When they were just map lookups that's not really a problem since its a fast, local operation. However, in this case you're calling `zkUtils.getLeaderAndIsrForPartition` twice, which requires making a request to zookeeper. And in fact, may return different results if something changes between the two calls.

Could we instead just call it once, store it into a variable of type `Option[LeaderAndIsr]` and then do the `isEmpty`/`get` on that variable?